### DISCUSSION TITLE
Provide tasmota-SML.bin which is pre-compiled with USE_SML_M on

### DISCUSSION THREAD

**Discussion by stefan123t** [arendst/Tasmota#17733] [Ideas]
Hi,
I am new to tasmota in general and I appreciate all the different language versions as well as 1M and 4M versions for different target platforms. I have only come by to program a standard ESP8266 (Wemos D1 mini pro for the fact) in order to use it with a so called Smart Meter Interface (eBZ DD3 BZ10 ETA - ODZ1) which can be read using a infrared TTL interface module.
I do not mind compiling / rolling my own tasmota build using the `user_config_override.h` as described here: https://tasmota.github.io/docs/Smart-Meter-Interface/
```c
#ifndef USE_SCRIPT
#define USE_SCRIPT
#endif
#ifndef USE_SML_M
#define USE_SML_M
#endif
#ifdef USE_RULES
#undef USE_RULES
#endif
```
But on the other hand I guess it would be rather simple to include a tasmota-SML or tasmota-4M-SML environment in the platformio.ini and roll these builds automatically. Especially with more and more users receiving a so called Smart Meter Interface in the basement and the Solar Energy vs. Grid Power consumption calculations this becomes a common use-case for tasmota.
Please let me know whether the suggested env and build are feasible and/or why there are objections against including the SML feature into a default build ?
Kind regards,
Stefan

**Comment by stefan123t** [comment]
Here is a video in german by someone who explains the process to roll your own SML binary, in case someone wonders
https://www.youtube.com/watch?v=s6qQs4FN9B0

Please note that it has already 100k views within a year, hence the need for a SML binary seems to be unquestionable.

**Comment by barbudor** [comment]
In additions of the official builds there are also additional builds here
- binaries https://github.com/Jason2866/Tasmota-specials/tree/firmware/firmware
- configurations : https://github.com/Jason2866/Tasmota-build/blob/special_builder/tasmota/user_config_override.h
- 
While they are mostly meant to be non-regression tests for the build process and not officially supported they are fully build from the same code base
May be one is already what you need, or close enough ans Jason's could accept a PR for a bit of enhancement

**Comment by Duvio** [comment]
I agree, there should be a pre-compiled release/build for this. Of course i could build it myself, but that way it would be way more convenient.

**Comment by Jason2866** [comment]
In the special builds the variant "scripting" includes `SML_M` https://github.com/Jason2866/Tasmota-build/blob/special_builder/tasmota/user_config_override.h#L1613

**Reply by Duvio** [reply]
It doesn't have rules, so it's not the same as https://tasmota.github.io/docs/Smart-Meter-Interface/. But I'll use it for now.

I recognize that your builds are not the official ones, but are you in any way affiliated with the Tasmota project?

**Reply by sfromis** [reply]
Jason is very much affiliated with the Tasmota project. And what was mentioned was indeed the smart meter interface, which is using scripting **instead of** having rules in the binary. Thus the "special" scripting build is likely what you ask for.

**Comment by DittelHome** [comment]
OK, the thread is a bit old, but the request is still there. Can we hope for a precompiled SML variant ?

**Reply by sfromis** [reply]
As Jason2866 mentioned, the scripting "unofficial" binary already includes USE_SML_M.
Found in https://github.com/tasmota/install/tree/firmware/firmware/unofficial

**Comment by DittelHome** [comment]
OK, sorry, I misunderstood that.
I use a ESP32 C3 (bitshake SmartMeter), therefore i have to build my own bin ?
I only find firmware/unofficial/tasmota32-scripting.bin.

**Reply by barbudor** [reply]
Correct
